src/jit/morph.cpp

```diff
@@ -8936,47 +8971,76 @@ GenTree* Compiler::fgMorphOneAsgBlockOp(GenTree* tree)
     //       [dest] [src]
     //
 
-    if (size == REGSIZE_BYTES)
+    if (asgType == TYP_STRUCT)
     {
-        if (clsHnd == NO_CLASS_HANDLE)
+        if (size == REGSIZE_BYTES)
```

Comment:
Move this to the switch below?

Reply:
I think it reads better this way, because otherwise you have to have all the `ifdef`s for the non-pointer-size cases.